src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiver.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.ons.ssdc.caseprocessor.messaging;
+
+import static uk.gov.ons.ssdc.caseprocessor.utils.MsgDateHelper.getMsgTimeStamp;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.springframework.integration.annotation.MessageEndpoint;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ssdc.caseprocessor.logging.EventLogger;
+import uk.gov.ons.ssdc.caseprocessor.model.dto.ResponseManagementEvent;
+import uk.gov.ons.ssdc.caseprocessor.model.dto.TelephoneCaptureDTO;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.Case;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.EventType;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.UacQidLink;
+import uk.gov.ons.ssdc.caseprocessor.service.CaseService;
+import uk.gov.ons.ssdc.caseprocessor.service.UacService;
+
+@MessageEndpoint
+public class TelephoneCaptureReceiver {
+
+  private final UacService uacService;
+  private final CaseService caseService;
+  private final EventLogger eventLogger;
+
+  private static final String TELEPHONE_CAPTURE_DESCRIPTION = "Telephone capture request received";
+
+  public TelephoneCaptureReceiver(
+      UacService uacService, CaseService caseService, EventLogger eventLogger) {
+    this.uacService = uacService;
+    this.caseService = caseService;
+    this.eventLogger = eventLogger;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "telephoneCaptureInputChannel")
+  public void receiveMessage(Message<ResponseManagementEvent> message) {
+    OffsetDateTime messageTimestamp = getMsgTimeStamp(message);
+    ResponseManagementEvent telephoneCaptureEvent = message.getPayload();
+    TelephoneCaptureDTO telephoneCapturePayload =
+        telephoneCaptureEvent.getPayload().getTelephoneCapture();
+
+    Case caze = caseService.getCaseByCaseId(telephoneCapturePayload.getCaseId());
+
+    // Double check the QID does not already exist
```

Comment:
Yeah should work now...needed to set events to null

Reply:
AH comment: "Factoring it out into a function would probably need a custom exception to achieve the same flow"

Reply:
Happy to refactor if you have a nicer pattern in mind but when I tried originally it ended up worse, since it has to be able to process, throw exception or just ignore the event.

Reply:
yeah fair enough - it's a bit annoying - but can live with it.

Reply:
I think it's fine. The code isn't more than a screenful and it's very simple